crates/native/src/user_media.rs

```diff
@@ -980,43 +980,200 @@ pub enum MediaTrackSource<T> {
     Remote { mid: String, peer: Weak<PeerConnection> },
 }
 
-/// Representation of a [`sys::VideoTrackInterface`].
-#[derive(AsRef)]
-pub struct VideoTrack {
-    /// ID of this [`VideoTrack`].
-    id: VideoTrackId,
+mod kind {
+    use std::sync::{Arc, OnceLock, RwLock};
 
-    /// Indicator whether this is a remote or a local track.
-    track_origin: TrackOrigin,
+    use derive_more::AsRef;
+    use libwebrtc_sys as sys;
 
-    /// Underlying [`sys::VideoTrackInterface`].
-    #[as_ref]
-    inner: sys::VideoTrackInterface,
+    use crate::{
+        AudioSource, AudioTrackId, VideoSink, VideoSinkId, VideoSource,
+        VideoTrackId, api,
+        frb_generated::StreamSink,
+        user_media::{
+            AudioLevelObserverId, AudioSourceAudioLevelHandler,
+            MediaTrackSource,
+        },
+    };
+
+    /// Representation of a [`sys::VideoTrackInterface`].
+    #[derive(AsRef)]
+    pub struct Video {
+        /// ID of this [`Video`].
+        pub id: VideoTrackId,
+
+        /// [`MediaTrackSource`] that is used by this [`Video`].
+        pub source: MediaTrackSource<VideoSource>,
+
+        /// Underlying [`sys::VideoTrackInterface`].
+        #[as_ref]
+        pub inner: sys::VideoTrackInterface,
```

Comment:
I really like the `pub inner`, but i guess this could be at least `pub(super) inner` :)